tests/cpp_unit_tests/test_dataset.cpp

```diff
@@ -443,7 +629,336 @@ TEST_CASE_TEMPLATE("Test dataset (common)", DatasetType, ConstDataset, MutableDa
                 }
             }
         }
+        SUBCASE("Homogeneous columnar buffer") {
+            SUBCASE("Single dataset") {
+                for (auto const elements_per_scenario : {0, 1, 2}) {
+                    CAPTURE(elements_per_scenario);
+                    auto const total_elements = elements_per_scenario;
+
+                    auto dataset = create_dataset(false, 1, dataset_type);
+
+                    auto id_buffer = std::vector<ID>(total_elements);
+                    auto a1_buffer = std::vector<double>(total_elements);
+
+                    add_homogeneous_buffer(dataset, A::name, elements_per_scenario, nullptr);
+
+                    auto const check_span = [&](auto const& buffer_span) {
+                        CHECK(buffer_span.size() == total_elements);
+                        for (Idx idx = 0; idx < buffer_span.size(); ++idx) {
+                            auto const element = [idx, &buffer_span] {
+                                if constexpr (std::same_as<DatasetType, ConstDataset>) {
+                                    static_assert(std::same_as<decltype(buffer_span[idx]),
+                                                               const_range_object<A::InputType const>::Proxy>);
+                                } else {
+                                    static_assert(std::same_as<decltype(buffer_span[idx]),
+                                                               mutable_range_object<A::InputType>::Proxy>);
+                                }
+                                return static_cast<A::InputType>(buffer_span[idx]);
+                            }();
+                            CHECK(element.id == id_buffer[idx]);
+                            CHECK(element.a1 == a1_buffer[idx]);
+                            CHECK(is_nan(element.a0));
+                        }
+                    };
```

Comment:
Sonar cloud
```suggestion
template <typename DatasetType, typename BufferSpan, typename Idx>
auto get_element(BufferSpan const& buffer_span, Idx idx) {
    if constexpr (std::same_as<DatasetType, ConstDataset>) {
        static_assert(std::same_as<decltype(buffer_span[idx]), const_range_object<A::InputType const>::Proxy>);
    } else {
        static_assert(std::same_as<decltype(buffer_span[idx]), mutable_range_object<A::InputType>::Proxy>);
    }
    return static_cast<A::InputType>(buffer_span[idx]);
}

auto const check_span = [&](auto const& buffer_span) {
    CHECK(buffer_span.size() == total_elements);
    for (Idx idx = 0; idx < buffer_span.size(); ++idx) {
        auto const element = get_element<DatasetType>(buffer_span, idx);
        CHECK(element.id == id_buffer[idx]);
        CHECK(element.a1 == a1_buffer[idx]);
        CHECK(is_nan(element.a0));
    }
};
```

Reply:
Thanks. Will test and resolve if passes SonarCloud.

Reply:
I had to remove the `static_assert` statements and refactor that, but now it passes.